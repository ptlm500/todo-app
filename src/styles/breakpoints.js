// width suggestions from https://www.freecodecamp.org/news/the-100-correct-way-to-do-css-breakpoints-88d6a5ba1862/
export default {
  phoneOnly: (content) => `
  @media (max-width: 599px) {
    ${content}
  }
  `,
  tabletPortaitUp: (content) => `
  @media (min-width: 600px) {
    ${content}
  }
  `,
  tabletLandscapeUp: (content) => `
  @media (min-width: 900px) {
    ${content}
  }
  `,
  desktopUp: (content) => `
  @media (min-width: 1200px) {
    ${content}
  }
  `,
  bigDesktopUp: (content) => `
  @media (min-width: 1800px) {
    ${content}
  }
  `
};
