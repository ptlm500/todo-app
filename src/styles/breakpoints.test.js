import breakpoints from './breakpoints';

const testContent = 'width: 100px';

describe('breakpoints', () => {
  it('exports a phoneOnly media query', () => {
    expect(breakpoints.phoneOnly(testContent)).toMatchSnapshot();
  });

  it('exports a tabletPortaitUp media query', () => {
    expect(breakpoints.tabletPortaitUp(testContent)).toMatchSnapshot();
  });

  it('exports a tabletLandscapeUp media query', () => {
    expect(breakpoints.tabletLandscapeUp(testContent)).toMatchSnapshot();
  });

  it('exports a desktopUp media query', () => {
    expect(breakpoints.desktopUp(testContent)).toMatchSnapshot();
  });

  it('exports a bigDesktopUp media query', () => {
    expect(breakpoints.bigDesktopUp(testContent)).toMatchSnapshot();
  });
});
