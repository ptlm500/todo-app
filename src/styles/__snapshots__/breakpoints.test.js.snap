// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breakpoints exports a bigDesktopUp media query 1`] = `
"
  @media (min-width: 1800px) {
    width: 100px
  }
  "
`;

exports[`breakpoints exports a desktopUp media query 1`] = `
"
  @media (min-width: 1200px) {
    width: 100px
  }
  "
`;

exports[`breakpoints exports a phoneOnly media query 1`] = `
"
  @media (max-width: 599px) {
    width: 100px
  }
  "
`;

exports[`breakpoints exports a tabletLandscapeUp media query 1`] = `
"
  @media (min-width: 900px) {
    width: 100px
  }
  "
`;

exports[`breakpoints exports a tabletPortaitUp media query 1`] = `
"
  @media (min-width: 600px) {
    width: 100px
  }
  "
`;
